packages/react-storage/src/components/StorageBrowser/Views/Controls/AddFiles.tsx

```diff
@@ -12,20 +15,58 @@ export interface AddFilesControl {
   (props: { disable: boolean }): React.JSX.Element;
 }
 
+function getItemsToAdd(items: LocationItem[], files: File[]) {
+  const locationItemsMap = new Map<string, LocationItem>();
+  items.forEach((item) => {
+    // If we have previously selected items, use them to create our map of unique items
+    const { key } = item;
+    locationItemsMap.set(key, item);
+  });
+
+  for (const data of files) {
+    const { name, lastModified, size } = data;
+
+    locationItemsMap.set(name, {
+      key: name,
+      data,
+      lastModified: new Date(lastModified),
+      size,
+      type: 'FILE',
+    });
+  }
+
+  return Array.from(locationItemsMap.values());
+}
```

Comment:
I think this could be a util for the `ActionSelect`, and `Add Folder` buttons too, but I'm not sure where to put it. Where should utils go?

Reply:
Let's keep it scoped to here for now

Reply:
Resolution: keep `Add Files` and `Add Folder` in the same file